tachyon/zk/base/prover_query.h

```diff
@@ -0,0 +1,34 @@
+#ifndef TACHYON_ZK_BASE_PROVER_QUERY_H_
+#define TACHYON_ZK_BASE_PROVER_QUERY_H_
+
+#include "tachyon/zk/base/polynomial_pointer.h"
+
+namespace tachyon::zk {
+
+template <typename PCSTy>
+class ProverQuery {
+ public:
+  using F = typename PCSTy::Field;
+  using Poly = typename PCSTy::Poly;
+
+  ProverQuery() = default;
+  ProverQuery(F& point, Poly& poly, F& blind)
+      : point_(point), poly_(poly), blind_(blind) {}
+
+  const F& point() const { return point_; }
```

Comment:
Please make getters for `poly_` and `blind_`. Or do you intentionally remove this since there's a `GetCommitment()`?

Reply:
Yes, it is intentional, but it would be good to have individual getters as well. I will add them.